src/PaymentGateways/Gateways/Stripe/Actions/CreatePaymentIntent.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe\Actions;
+
+use Give\LegacyPaymentGateways\Adapters\LegacyDonationData;
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\DonationSummary;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentIntent;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentMethod;
+use Give\PaymentGateways\Gateways\Stripe\WorkflowAction;
+use Give\ValueObjects\Money;
+
+class CreatePaymentIntent extends WorkflowAction
+{
+    public function __invoke(
+        GatewayPaymentData $paymentData,
+        DonationSummary $donationSummary,
+        \Give_Stripe_Customer $giveStripeCustomer,
```

Comment:
@kjohnson I agree that you about less rewrite code for gateway migration. Any specific reason to use a legacy class? Can we handle it differently in which we do not require legacy classes?

Reply:
There is a lot that I _could_ refactor, which I avoided in this PR in order to limit the scope. The goal here is to update the gateway to use the new Gateway API. A part of that is making new code testable, so I made some additional refactors to support the primary goal. Some aspects were skipped for sake of capacity.

In this case, I was able to mock `Give_Stripe_Customer`, so I left it as is.